src/amoc_api_scenarios_handler.erl

```diff
@@ -10,112 +10,129 @@
          allowed_methods/2,
          content_types_provided/2,
          content_types_accepted/2,
-         process_request/2,
-         process_json/2]).
+         to_json/2,
+         from_json/2]).
 
--record(state, {action}).
+
+-record(state, {module_name, module_source}).
 
 -type state() :: #state{}.
 
--spec trails() -> [tuple()] .
+-spec trails() -> [trails:trail()] .
 trails() ->
     Metadata =
-        #{get =>
-          #{tags => ["scenarios"],
-            description => "Lists available scenarios",
-            produces => ["application/json"]
-          },
-          post =>
-          #{tags => ["scenarios"],
-            description => "Adds new scenario",
-            produces => ["application/json"]
-          }
-    },
-    [trails:trail("/scenarios", amoc_api_scenarios_handler, [], Metadata)].
-
--spec init(tuple(), cowboy:req(), state()) -> {upgrade, protocol, cowboy_rest}.
+    #{get =>
+      #{tags => ["scenarios"],
+        description => "Gets list of available scenarios",
+        produces => ["application/json"]
+       },
+      post =>
+      #{tags => ["scenarios"],
+        description => "Uploads new scenario",
+        produces => ["application/json"]
+       }
+     },
+    [trails:trail("/scenarios", ?MODULE, [], Metadata)].
+
+-spec init(tuple(), cowboy_req:req(), state()) -> {upgrade, protocol, cowboy_rest}.
```

Comment:
Line 37 is too long: -spec init(tuple(), cowboy_req:req(), state()) -> {upgrade, protocol, cowboy_rest}..